internal/pkg/docker/docker.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package docker
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/ecr"
+)
+
+// TODO: wrap the `os/exec` stuff in a `command` package to make mocking and testing easier on this package.
+
+// Service wraps a repository URI endpoint.
+type Service struct {
+	repositoryURI string
+}
+
+// New returns a Service configured with the input URI.
+func New(uri string) Service {
+	return Service{
+		repositoryURI: uri,
+	}
+}
+
+// Build will `os/exec` a `docker build` command with the input tag and Dockerfile image path.
+func (s Service) Build(imageTag, path string) error {
+	imageName := fmt.Sprintf("%s:%s", s.repositoryURI, imageTag)
+
+	cmd := exec.Command("docker", "build", "-t", imageName, path)
+	// TODO: investigate if setting cmd.Stdout is needed
+	// cmd.Stdout = os.Stderr
```

Comment:
If we do this, do we get the stream events that's happening while building the image written to `stderr`? If so that seems like a nice change.

Otherwise, I believe now we only write the image events post-build

Reply:
What I left behind here is actually invalid, but the only way I've found to output the `docker build` output is to capture it from `cmd.Output()` and then write it to `stderr` explicitly. I think we want to expose this info so `docker build`s are clear to the user and won't interfere with piping.

Reply:
yeah I agree that we should expose it, would this help:
https://github.com/aws/amazon-ecs-cli/blob/12b3d91596b92e8790cdc88d4a7b2c84dadc1db4/ecs-cli/modules/cli/local/up_app.go#L222-L232

Reply:
The nuance turns out to be this:
```
// Stdout and Stderr specify the process's standard output and error.
//
// If either is nil, Run connects the corresponding file descriptor
// to the null device (os.DevNull).
```